src/blot/inline.ts

```diff
@@ -12,12 +12,12 @@ import LeafBlot from './abstract/leaf';
 import ParentBlot from './abstract/parent';
 
 // Shallow object comparison
-function isEqual(obj1: object, obj2: object): boolean {
+type Obj = Record<string, unknown>;
```

Comment:
I'd avoid defining a type for `Record<string, unkown>` as `Obj` is meaningless.

Reply:
Yeah, it's only for making TS happy
<img src="https://github.com/quilljs/parchment/assets/5557143/2bbf557f-c266-48e6-b8d8-3280d1492328" width="500" /> 
though I think the ts error is correct because object index types should be limited, otherwise anything is allowed 

<img src="https://github.com/quilljs/parchment/assets/5557143/aed05a4d-92e9-4153-b262-f62e5bd68314" width="300" /> 

And the check prevents that
<img src="https://github.com/quilljs/parchment/assets/5557143/84738168-dcce-4fd5-90fb-d61801826265" width="500" /> 

...though in runtime it doesn't really matter
<img src="https://github.com/quilljs/parchment/assets/5557143/9a8162ef-848a-4719-8db5-3fb95111f554" width="400" /> 
but I think it's still nice to have the type check

Reply:
Actually I mean we can inline the type:

```ts
function isEqual(
  obj1: Record<string, unknown>,
  obj2: Record<string, unknown>,
): boolean {
```

We don't need to give `Record<string, unknown>` an alias unless the alias is meaningful in the context.